backend/layers/processing/reprocess_dataset_metadata.py

```diff
@@ -114,11 +114,22 @@ def update_rds(
 
         rds_object = base.readRDS(seurat_filename)
 
+        new_keys = []
+        seurat_metadata = seurat.Misc(object=rds_object)
         for key, val in metadata_update.as_dict_without_none_values().items():
-            seurat_metadata = seurat.Misc(object=rds_object)
             if seurat_metadata.rx2[key]:
                 val = val if isinstance(val, list) else [val]
                 seurat_metadata[seurat_metadata.names.index(key)] = StrVector(val)
+            else:
+                new_keys.append((key, val))
+
+        if new_keys:
+            new_key_vector = ListVector({k: v for k, v in new_keys})
+            seurat_metadata += new_key_vector
+            r.assign("rds_object", rds_object)
+            r.assign("seurat_metadata", seurat_metadata)
+            r("rds_object@misc <- rds_misc")
+            rds_object = r["rds_object"]
```

Comment:
Not totally clear to me what's going on here

Reply:
rpy2 is an interface for running R in a Python process. It provides a number of functions / objects that allow you to perform R operations pythonically, but it doesn't integrate well with all R libraries. So, it also provides an interface that lets you write R directly. 

With Seurat, I found that the rpy2 python representation of the 'Misc' dict (this is the Seurat 'uns' equivalent) doesn't support adding new fields (if you tried to saveRDS after the transformation above, it does not persist the additions). Though it does persist editing values of existing keys (existing code). 

So, this block adds the new metadata to the Misc object, assigns the python objects to R objects, and runs the R code to update the Misc dict for the rds_object. Then passes it back into a python object to be saved below.

Reply:
Fair enough!